tests/S-4_Account_creation _deletion_and _get_info.test.js

```diff
@@ -0,0 +1,369 @@
+
+    // Iterate over each username
+    for (const userName of usernames) {
+        // Iterate over each className
+        for (const className of classNames) {
+            // Make the GET request
+            const response = await t.context.got.get(`user/${userName}/class/${className}`);
+            
+            // Check status Code
+            t.is(response.statusCode, 200, `Expected status code 200 for ${userName} and ${className}`);
+
+            // Assign the response to the pre-allocated array
+            responses[responseIndex++] = response;
+
+            // Check Response body once for each className (since we check the other later against thoses ones)
+            if (userName == usernames[0]){
+
+                var body = responses[responseIndex-1].body;
+                // Check that the first response has the expected body
+                
+                // Make sure it has a className and it is the expected one
+                t.not(body.className, undefined, "Response should have a className property");
+                t.is(body.className, className, `Expected className to be equal with that of the request ${className}`);
+                
+                // Make sure it is an array and that it has a length
+                t.true(Array.isArray(body.users), "Users should be an array");
+                t.not(body.users.length,undefined, "Expected to be able to get lenght of users array");
+                
+                // Loop over all users in the array
+                for (var i = 0 ; i < body.users.length ; i++ ){
+                    const dummy_user = body.users[i];
+                    // Check that the struct has the proper fields  
+                    t.not(dummy_user.grade,          undefined, "User should have a grade property");
+                    t.not(dummy_user.user,           undefined, "User should have a user property");
+                    t.not(dummy_user.user.userName,  undefined, "User should have a userName property");
+                    t.not(dummy_user.user.surname,   undefined, "User should have a surname property");
+                    t.not(dummy_user.user.name,      undefined, "User should have a name property");
+                    t.not(dummy_user.user.id,        undefined, "User should have a id property");
+                    t.not(dummy_user.user.email,     undefined, "User should have a email property");        
+                }
+            }
+        }
+    }
+
+    // Check that all responses have the same body for each class
+    for (let i = 1; i < responses.length; i++) {
+        if(i == i%classNames.length ) {continue;}
+        t.deepEqual(
+            responses[i].body,
+            responses[i%classNames.length].body,
+            `Expected the same body response as ${usernames[i%classNames.length]} and ${classNames[i%classNames.length]} for ${usernames[Math.floor(i / classNames.length)]} and ${classNames[i % classNames.length ]}`
```

Comment:
I would have chosen this checking to be inside of the double for-loop, to avoid the complicated index arithmetic.
We'd have to create a `classes` object, store the response using `classes[className] = response`, and if  `classes[className]` already exists check that the response is equal to it.

Reply:
I made a new commit to fix this